modules/network/src/main/java/org/apache/ignite/internal/network/netty/NettyUtils.java

```diff
@@ -38,18 +41,21 @@ public class NettyUtils {
      * @return CompletableFuture.
      */
     public static <T, R, F extends Future<R>> CompletableFuture<T> toCompletableFuture(
-            F nettyFuture,
+            @Schedule F nettyFuture,
             Function<F, T> mapper
     ) {
         var fut = new CompletableFuture<T>();
 
-        nettyFuture.addListener((F future) -> {
-            if (future.isSuccess()) {
-                fut.complete(mapper.apply(future));
-            } else if (future.isCancelled()) {
-                fut.cancel(true);
-            } else {
-                fut.completeExceptionally(future.cause());
+        nettyFuture.addListener(new GenericFutureListener<F>() {
+            @Override
+            public void operationComplete(@Execute F future) throws Exception {
```

Comment:
Is it impossible to add annotations to lambda parameters?

Reply:
TIL that it is possible